datafusion/sqllogictest/test_files/timestamps.slt

```diff
@@ -331,6 +331,35 @@ SELECT COUNT(*) FROM ts_data_secs where ts > to_timestamp_seconds('2020-09-08T12
 ----
 2
 
+# to_timestamp with formatting
+query I
+SELECT COUNT(*) FROM ts_data_nanos where ts > to_timestamp('2020-09-08T12:00:00+00:00', '2020-09-08 12/00/00+00:00', '%c', '%+', '%Y-%m-%d %H/%M/%s%#z')
```

Comment:
Can we also please add some tests that show

1.  `to_timestamp` with formatting being invoked on a table with a string column (not just constant values)
2. error case for bad format strings
3. error cases where some of the values in a column can be parsed with the format string, but some can not

Reply:
#2 is primarily covered in lines 1992 - 2010. I'll update the tests to include #1 & #3 and #2 using a table vs static string.